src/main/java/uk/gov/register/serialization/CommandParser.java

```diff
@@ -75,17 +80,18 @@ public void addCommand(String s) {
                 break;
             case "append-entry":
                 if (parts.length == 4) {
-                    Entry entry = new Entry(0, stripPrefix(parts[2]), Instant.parse(parts[1]), parts[3]);
+                    Entry entry = new Entry(0, new HashValue(HashingAlgorithm.SHA256, stripPrefix(parts[2])), Instant.parse(parts[1]), parts[3]);
                     entries.put(position++, entry);
-                    updateItemHashCount(entry.getSha256hex());
+                    updateItemHashCount(entry.getSha256hex().getValue());
                 } else {
                     LOG.error("append entry line must have 4 elements, was : " + s);
                     throw new SerializedRegisterParseException("append entry line must have 4 elements, was : " + s);
                 }
                 break;
             case "assert-root-hash":
                 if (parts.length == 2) {
-                    RegisterProof registerProof = new RegisterProof(parts[1]);
+                    HashValue hash = HashValue.decode(HashingAlgorithm.SHA256.toString(), parts[1]);
+                    RegisterProof registerProof = new RegisterProof(hash.getValue());
```

Comment:
`RegisterProof` should take a `HashValue` in its constructor rather than a `String` and `RegisterProof.rootHash` property should be of type `HashValue`.